torch/csrc/jit/tensorexpr/operators/conv2d.cpp

```diff
@@ -0,0 +1,83 @@
+#include <torch/csrc/jit/tensorexpr/operators/conv2d.h>
+#include <torch/csrc/jit/tensorexpr/tensor.h>
+
+namespace torch::jit::tensorexpr {
+
+Tensor* conv2d_depthwise(
+    BufHandle input,
+    BufHandle weight,
+    BufHandle bias,
+    int stride,
+    int pad,
+    int groups) {
+  TORCH_INTERNAL_ASSERT(input.ndim() == 4);
+  TORCH_INTERNAL_ASSERT(weight.ndim() == 4);
+
+  auto const& N = immediateAs<int>(input.dim(0));
```

Comment:
Should we check if dims are actually immediates or are we ok with an assertion in `immediateAs<int>`?

Reply:
yeah, good point, `immediateAs` will give a really opaque error message like "UNSUPPORTED DTYPE" (which is maybe a separate thing to fix but not quite the fault of this diff :-p ).  I'll add an explicit check.